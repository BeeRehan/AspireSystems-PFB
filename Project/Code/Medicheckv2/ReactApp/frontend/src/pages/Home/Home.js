import {React} from 'react'

import Carousell  from '../../components/universe/Carousell.js';
import HomeNav from '../../components/universe/HomeNav.js';

export default function Home() {
    return (
      <>
        <HomeNav/>
        <Carousell/>
      </>
    )
}