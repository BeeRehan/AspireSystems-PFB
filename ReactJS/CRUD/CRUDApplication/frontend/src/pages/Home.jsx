import {Button,Form,FloatingLabel,} from 'react-bootstrap'
import {React,useState,useRef} from 'react'
import {useNavigate} from 'react-router-dom'
import { useForm } from 'react-hook-form';
import { yupResolver } from '@hookform/resolvers/yup';
import * as yup from 'yup';
import ReCAPTCHA from "react-google-recaptcha";

const schema = yup.object().shape({
    name: yup.string().required(),
    age: yup.number().required().positive().integer(),
    occupation: yup.string().required().min(8).max(24),

}).required();


export default function Home() {

  const [details,setDetails] = useState({});
  const navigate = useNavigate();
  const refCaptcha = useRef(null);
  
  const { register, handleSubmit, formState: { errors } } = useForm({
    resolver: yupResolver(schema),
  });

  function handler(e){
    setDetails((prev)=>{
    return{
      ...prev,
    [e.target.name] : e.target.value      
    };
      
    })
  }

function validateCaptcha(){
  fetch('http://localhost:8000/api/rest-create/',{
    method:"POST",
    headers:{
      "Content-type":"application/json",
    },
    body:JSON.stringify(details)
  }).then((res)=>{
    if(res.ok){
      console.log("Pushed");
      navigate("/list")
    }
  }).catch(er=>{
    console.log("Error:",er);
  }) 
  }

  async function submit(e){
    console.log("Details:",details);
    const capToken = await refCaptcha.current.executeAsync();
    console.log("Token",capToken)
    refCaptcha.current.reset()
    const key = '6Lcg0cEdAAAAAFTNpNMC6C_z_J89wC7Ydav9DhUA'
    fetch(`https://www.google.com/recaptcha/api/siteverify?secret=${key}&response=${capToken}`,{
     method:'POST',
   }).then((res)=>{
     if(res.ok){
       return res.json()
     }
     return res
   }).then(res=>{
      if(res.success){
        validateCaptcha()
      }
   }).catch(err=>{
     console.log(err);
   })

    return false;
  }

  return (
    <div>
      <form className="form" onSubmit={handleSubmit(submit)}>
        <FloatingLabel controlId="floatingInputName"  label="Name" className="mb-3">
          <Form.Control type="text" placeholder="name" {...register('name', { required: true })} name="name" value={details.name} onChange={handler} required/>
          <span>{errors.name?.message}</span>
        </FloatingLabel>
        <FloatingLabel controlId="floatingInputAge" label="Age" className="mb-3">
          <Form.Control type="number" placeholder="age" {...register('age', { required: true })} name="age" value={details.age} onChange={handler} required/>
          <span>{errors.age?.message}</span>
        </FloatingLabel>
        <FloatingLabel controlId="floatingInputOccupation" label="Occupation" className="mb-3">
          <Form.Control type="text" placeholder="occupation" {...register('occupation', { required: true })} name="occupation" value={details.ooccupation} onChange={handler} required/>
          <span>{errors.occupation?.message}</span>
        </FloatingLabel>
        <ReCAPTCHA ref={refCaptcha} sitekey="6Lcg0cEdAAAAABTc9I-jLzQ2SoE9243yMPxKOBtq" size='invisible'/>
        <Button type="submit">Submit</Button>
      </form>
    </div>
  )
}