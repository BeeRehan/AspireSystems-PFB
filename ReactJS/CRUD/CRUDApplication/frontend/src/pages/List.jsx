import {Button,Table,Container} from 'react-bootstrap'
import {React,useState,useEffect} from 'react'
import {Link} from 'react-router-dom'

import Skeleton from '@material-ui/lab/Skeleton';

export default function List() {
    const [users, setUsers] = useState(null);
    function fetchData(){
      fetch('http://localhost:8000/api/rest-list/')
      .then(res=>{
        if(res.ok){
          return(
            res.json()
          ); 
        }
        throw res
      })
      .then(data=>{
        // console.log("DataR",data)
        setTimeout(() => {
          setUsers(data)
        }, 2021);
      })
      .catch(er=>{
        console.log("Error in List",er);
      })
    }
  
    useEffect(() => {
      fetchData()
    },[])
  
    function toDelete(id){
      console.log("Deleted");
      fetch(`http://localhost:8000/api/rest-delete/${id}`,{
        method:"DELETE"
      })
      .then(res=>{
        if(res.ok){
          fetchData()
          // window.location.reload()
        }
        throw res
      })
      .catch(er=>{
        console.log("Error in List",er);
      })
      
    }
    console.log("user",users)
    if(users){
      return (
        <Container>
          <Table striped bordered hover>
          <thead>
            <tr>
                <th>Name</th>
                <th>Age</th>
                <th>Occupation</th>
                <th>To Edit</th>
                <th>To Delete</th>
            </tr>
            </thead>
            <tbody>
              {
                users.map((user)=>{
                  return(
                    <tr key={user.id}>
                    <td>{user.name}</td>
                    <td>{user.age}</td>
                    <td>{user.occupation}</td>
                    <td><Link to={`/edit/${user.id}`}>Edit</Link></td>
                    <td><Button onClick={()=>toDelete(user.id)}>Delete</Button></td>
                    </tr>
                  );
                })
              }
            </tbody>
          </Table>
          <Button as={Link} to={'/'}>Add new </Button>
        </Container>
      ) 
    }
    else{
      return(
        <Container>
        <Table striped bordered hover>
        <thead>
          <tr>
              <th>Name</th>
              <th>Age</th>
              <th>Occupation</th>
              <th>To Edit</th>
              <th>To Delete</th>
          </tr>
          </thead>
          <tbody>
            {
              [1,2,3].map((user)=>{
                return(
                  <tr key={user}>
                  <td><Skeleton /></td>
                  <td><Skeleton animation="wave" /></td>
                  <td><Skeleton animation="wave" /></td>
                  <td><Skeleton animation="wave" /></td>
                  <td><Skeleton animation="false" /></td>
                  </tr>
                );
              })
            }
          </tbody>
        </Table>
        <Button as={Link} to={'/'}>Add new </Button>
      </Container>
        
      );
    }
  }