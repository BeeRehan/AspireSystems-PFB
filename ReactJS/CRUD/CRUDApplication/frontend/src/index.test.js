import { render, screen } from '@testing-library/react';
import { shallow, mount } from "enzyme";
import toJSON from 'enzyme-to-json'
import Home from './pages/Home';
import List from './pages/List';

test('to test home compounent',()=>{
    render(<Home/>);
    const button = screen.getByRole('button', {
        name: /submit/i
      });
      expect(button).toBeInTheDocument();
})


it("renders without crashing", () => {
  const base = shallow(<Home />);
  const button = <h1>Add Here</h1>
  expect(base.contains(button)).not.toEqual(true);

});

it("renders correctly", () => {
  const tree = shallow(<Home/>);
  expect(toJSON(tree)).toMatchSnapshot();
});
