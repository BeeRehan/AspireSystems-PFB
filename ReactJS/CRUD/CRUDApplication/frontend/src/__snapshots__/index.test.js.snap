// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Container
  fluid={false}
>
  <ForwardRef
    bordered={true}
    hover={true}
    striped={true}
  >
    <thead>
      <tr>
        <th>
          Name
        </th>
        <th>
          Age
        </th>
        <th>
          Occupation
        </th>
        <th>
          To Edit
        </th>
        <th>
          To Delete
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        key="1"
      >
        <td>
          <WithStyles(ForwardRef(Skeleton)) />
        </td>
        <td>
          <WithStyles(ForwardRef(Skeleton))
            animation="wave"
          />
        </td>
        <td>
          <WithStyles(ForwardRef(Skeleton))
            animation="wave"
          />
        </td>
        <td>
          <WithStyles(ForwardRef(Skeleton))
            animation="wave"
          />
        </td>
        <td>
          <WithStyles(ForwardRef(Skeleton))
            animation="false"
          />
        </td>
      </tr>
      <tr
        key="2"
      >
        <td>
          <WithStyles(ForwardRef(Skeleton)) />
        </td>
        <td>
          <WithStyles(ForwardRef(Skeleton))
            animation="wave"
          />
        </td>
        <td>
          <WithStyles(ForwardRef(Skeleton))
            animation="wave"
          />
        </td>
        <td>
          <WithStyles(ForwardRef(Skeleton))
            animation="wave"
          />
        </td>
        <td>
          <WithStyles(ForwardRef(Skeleton))
            animation="false"
          />
        </td>
      </tr>
      <tr
        key="3"
      >
        <td>
          <WithStyles(ForwardRef(Skeleton)) />
        </td>
        <td>
          <WithStyles(ForwardRef(Skeleton))
            animation="wave"
          />
        </td>
        <td>
          <WithStyles(ForwardRef(Skeleton))
            animation="wave"
          />
        </td>
        <td>
          <WithStyles(ForwardRef(Skeleton))
            animation="wave"
          />
        </td>
        <td>
          <WithStyles(ForwardRef(Skeleton))
            animation="false"
          />
        </td>
      </tr>
    </tbody>
  </ForwardRef>
  <Button
    active={false}
    as={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "render": [Function],
      }
    }
    disabled={false}
    to="/"
    variant="primary"
  >
    Add new 
  </Button>
</Container>
`;
